src/test/java/edu/unc/lib/boxc/migration/cdm/services/ArchivalDestinationsServiceTest.java

```diff
@@ -138,4 +139,129 @@ public void collectionNumberToPidTest() throws Exception {
+        }
+    }
+
+    @Test
+    public void newDestinationsFileWithoutPid() throws Exception {
+        solrResponseWithoutPid();
+        Path destinationMappingsPath = project.getDestinationMappingsPath();
+
+        var options = new DestinationMappingOptions();
+        options.setFieldName("dmrecord");
+
+        service.addArchivalCollectionMappings(options);
+
+        try (
+                Reader reader = Files.newBufferedReader(destinationMappingsPath);
+                CSVParser csvParser = new CSVParser(reader, CSVFormat.DEFAULT
+                        .withFirstRecordAsHeader()
+                        .withHeader(FieldUrlAssessmentService.URL_CSV_HEADERS)
+                        .withTrim());
+        ) {
+            List<CSVRecord> rows = csvParser.getRecords();
+            assertEquals("", rows.get(0).get(1));
+            assertEquals("", rows.get(1).get(1));
+        }
+    }
+
+    @Test
+    public void existingDestinationsFileWithoutForceFlag() throws Exception {
+        solrResponseWithPid();
+        writeCsv(mappingBody("aid:40126,bdbd99af-36a5-4bab-9785-e3a802d3737e,"));
+
+        var options = new DestinationMappingOptions();
+        options.setFieldName("dmrecord");
+
+        try {
+            service.addArchivalCollectionMappings(options);
+            fail();
+        } catch (StateAlreadyExistsException e) {
+            assertTrue(e.getMessage().contains("Cannot create destinations, a file already exists."
+                    + " Use the force flag to overwrite."));
+        }
+    }
+
+    @Test
+    public void existingDestinationsFileWithForceFlag() throws Exception {
+        solrResponseWithPid();
+        writeCsv(mappingBody("aid:40126,bdbd99af-36a5-4bab-9785-e3a802d3737e,"));
+
+        var options = new DestinationMappingOptions();
+        options.setFieldName("dmrecord");
+        options.setForce(true);
+
+        service.addArchivalCollectionMappings(options);
+
+    }
+
+    @Test
+    public void archivalCollectionMappingsWithoutFieldOption() throws Exception {
+        var options = new DestinationMappingOptions();
+
+        try {
+            service.addArchivalCollectionMappings(options);
+            fail();
+        } catch (Exception e){
+            assertTrue(e.getMessage().contains("Field option is empty"));
+        }
+    }
+
+    private void solrResponseWithoutPid() throws Exception {
+        testHelper.indexExportData("mini_keepsakes");
+        QueryResponse testResponse = new QueryResponse();
+        testResponse.setResponse(new NamedList<>(Map.of("response", new SolrDocumentList())));
+        when(solrClient.query(solrQueryCaptor.capture())).thenReturn(testResponse);
+    }
+
+    private void solrResponseWithPid() throws Exception {
+        testHelper.indexExportData("mini_keepsakes");
+
+        QueryResponse testResponseA = new QueryResponse();
+        SolrDocument testDocumentA = new SolrDocument();
+        testDocumentA.addField("pid", "40147");
```

Comment:
Sorry I didn't catch this early, I'm realizing that the solr document field here doesn't match up with what is in boxc. This class in boxc lists the solr fields:
https://github.com/UNC-Libraries/box-c/blob/main/search-api/src/main/java/edu/unc/lib/boxc/search/api/SearchFieldKey.java#L14
And is available in chompb, so I'd suggest using it as constants in all the cases where we reference a solr field in chompb, both in responses and when generating the query. So something like `SearchFieldKey.ID.getSolrField()` and `SearchFieldKey.COLLECTION_ID.getSolrField()`. It'll help keep things in sync across applications too. We'll need to update `generateCollectionNumbersToPidMapping` as well.

Also, the `id` field would have values like `2720138a-a1ab-4b73-b7bc-6aabe75620c0`, while realistically `40147` is the kind of value we'd find in the `collectionId` field (although in this test it is using contentdm ids, which is fine for testing), so I'd suggest returning a UUID form value here to avoid confusion (I was pretty confused when reading some of the assertions in tests)